File: C++/test/_00151_ReverseWordsInAString_test.cpp
#include <gtest/gtest.h>
#include "../src/_00151_ReverseWordsInAString.cpp"

TEST(ReverseWordsInAString, Example1) {
    Solution sol;

    std::string input = "the sky is blue";
    std::string expected = "blue is sky the";

    EXPECT_EQ(sol.reverseWords(input), expected);
}

TEST(ReverseWordsInAString, Example2) {
    Solution sol;

    std::string input = "  hello world ";
    std::string expected = "world hello";

    EXPECT_EQ(sol.reverseWords(input), expected);
}

TEST(ReverseWordsInAString, Example3) {
    Solution sol;

    std::string input = "a good   example";
    std::string expected = "example good a";

    EXPECT_EQ(sol.reverseWords(input), expected);
}

File: JavaScript/test/_00151_ReverseWordsInAString.test.js
const reverseWords = require("../src/_00151_ReverseWordsInAString");

test("Example 1", () => {
    expect(reverseWords("the sky is blue")).toBe("blue is sky the");
});

test("Example 2", () => {
    expect(reverseWords("  hello world  ")).toBe("world hello");
});

test("Example 3", () => {
    expect(reverseWords("a good   example")).toBe("example good a");
});

test("One word returns word", () => {
    expect(reverseWords("word")).toBe("word");
})
